### PR TITLE
Request and report cmd

### LINKED ISSUE #527: Design and implement CLI for sending request and report transaction
Let's talk about the design with @sorawit first.

### PR DESCRIPTION
**Pull request by taobun**
fixed #527

### CODE REVIEW

File: chain/x/zoracle/client/cli/tx.go [hunk 1]
```diff
@@ -97,14 +109,32 @@ func GetCmdRequest(cdc *codec.Codec) *cobra.Command {
...
+	return cmd
 }
 
 // GetCmdReport implements the report command handler
 func GetCmdReport(cdc *codec.Codec) *cobra.Command {
 	return &cobra.Command{
-		Use:   "report [requestid] [data]",
-		Short: "report data",
-		Args:  cobra.ExactArgs(2),
+		Use:   "report [requestid] ([data]...)",
```
Comment (sorawit) [inline chain/x/zoracle/client/cli/tx.go]: requestID

File: chain/x/zoracle/client/cli/tx.go [hunk 2]
```diff
@@ -97,14 +109,32 @@ func GetCmdRequest(cdc *codec.Codec) *cobra.Command {
...
+		Short: "Report to given request id",
+		Args:  cobra.MinimumNArgs(2),
+		Long: strings.TrimSpace(
+			fmt.Sprintf(`Report to unresolved request, need to report data equal to the number of raw data request
```
Comment (sorawit) [inline chain/x/zoracle/client/cli/tx.go]: Report to an unresolved request. All raw data requests must be reported at once.

File: chain/x/zoracle/client/cli/tx.go [hunk 3]
```diff
@@ -46,10 +50,18 @@ func GetTxCmd(storeKey string, cdc *codec.Codec) *cobra.Command {
...
-	return &cobra.Command{
-		Use:   "request [oracleScriptID] [calldata] [requestedCount] [sufficientValidator] [Expiration]",
-		Short: "request open api data",
-		Args:  cobra.ExactArgs(5),
+	cmd := &cobra.Command{
+		Use:   "request [oracleScriptID] (--calldata [calldata]) (--require-validator [requestedValidatorCount]) (--sufficient-validator [sufficientValidatorCount]) (--expiration [Expiration])",
+		Short: "Request a new request from an existed oracle script",
+		Args:  cobra.ExactArgs(1),
+		Long: strings.TrimSpace(
+			fmt.Sprintf(`Create new request from an existed oracle script with configure flags.
```
Comment (sorawit) [inline chain/x/zoracle/client/cli/tx.go]: an existing ... with the

File: chain/x/zoracle/client/cli/tx.go [hunk 4]
```diff
@@ -116,11 +146,23 @@ func GetCmdReport(cdc *codec.Codec) *cobra.Command {
...
+					return fmt.Errorf("Invalid report format: %s", reportRaw[0])
+				}
+				externalID, err := strconv.ParseInt(reportRaw[0], 10, 64)
+				if err != nil {
+					return err
+				}
+				dataset = append(dataset, types.NewRawDataReport(externalID, []byte(reportRaw[1])))
 			}
 
+			// Sort report by external ID
```
Comment (sorawit) [inline chain/x/zoracle/client/cli/tx.go]: Sort data reports

File: chain/x/zoracle/client/cli/tx.go [hunk 5]
```diff
@@ -19,10 +19,14 @@ import (
...
-	flagName    = "name"
-	flagScript  = "script"
-	flagCallFee = "call-fee"
-	flagOwner   = "owner"
+	flagName                = "name"
+	flagScript              = "script"
+	flagCallFee             = "call-fee"
+	flagOwner               = "owner"
+	flagCalldata            = "calldata"
+	flagRequireValidator    = "require-validator"
```
Comment (sorawit) [inline chain/x/zoracle/client/cli/tx.go]: RequestedValidatorCount rvc

File: chain/x/zoracle/client/cli/tx.go [hunk 6]
```diff
@@ -19,10 +19,14 @@ import (
...
+	flagSufficientValidator = "sufficient-validator"
```
Comment (sorawit) [inline chain/x/zoracle/client/cli/tx.go]: SufficientValidatorCount svc

File: chain/x/zoracle/client/cli/tx.go [hunk 7]
```diff
@@ -19,10 +19,14 @@ import (
...
+	flagExpiration          = "expiration"
```
Comment (sorawit) [inline chain/x/zoracle/client/cli/tx.go]: Expiration exp

**Review by sorawit — CHANGES_REQUESTED** [review]

File: chain/x/zoracle/client/cli/tx.go [hunk 8]
```diff
@@ -46,10 +50,18 @@ func GetTxCmd(storeKey string, cdc *codec.Codec) *cobra.Command {
 
 // GetCmdRequest implements the request command handler
 func GetCmdRequest(cdc *codec.Codec) *cobra.Command {
-	return &cobra.Command{
-		Use:   "request [oracleScriptID] [calldata] [requestedCount] [sufficientValidator] [Expiration]",
-		Short: "request open api data",
-		Args:  cobra.ExactArgs(5),
+	cmd := &cobra.Command{
+		Use:   "request [oracleScriptID] (--calldata [calldata]) (--require-validator [requestedValidatorCount]) (--sufficient-validator [sufficientValidatorCount]) (--expiration [Expiration])",
```
Comment (sorawit) [inline chain/x/zoracle/client/cli/tx.go]: use lowercase [expiration]. Be consistent!

**Review by sorawit — CHANGES_REQUESTED** [review]

**Review by sorawit — APPROVED** [review]

**Review by prin-r — APPROVED** [review]

**Pull request merged**